### ISSUE TITLE
Should we allow linking AudioSessions to each HTMLMediaElement/AudioContext object?

### ISSUE THREAD

**Issue by youennf** [youennf/audio-focus#5]
Is it useful to set a particular AudioSession to a particular HTMLMediaElement/AudioContext?

**Comment by jernoble** [comment]
I think it is. A site may have both alert sounds ("transient") and a music stream ("playback"). It would be convenient to allow alert sounds to duck playback the same way a system alert sound would, without requiring the page to "re-invent the wheel" and implement ducking manually.

**Comment by jernoble** [comment]
The most straightforward API design would involve monkey-patching HTML to add a `.audioSession` property on HTMLMediaElement and AudioContext. I don't know how this would work for `navigator.mediaDevices.getUserMedia()` however. Maybe `navigator.mediaDevices.getUserMedia({audio: { audioSession: captureSession }})`?

**Comment by jernoble** [comment]
Alternatively, the `MediaStream` returned by `getUserMedia()` could have a `.audioSession` property. But we'd need a the `MediaStream` to start out deactivated.

**Comment by youennf** [comment]
Right, this would require such setters/getters, plus AudioSession constructor, plus requestFocus/releaseFocus.
This seems potentially useful as a second step.

**Comment by youennf** [comment]
For instance
```javascript
partial interface AudioSession {
  constructor();

  // Request audio focus from the platform. Resolves with true if the request was successful.
  Promise<bool> request();

  // Abandons audio focus. Rejects with an error if the session does not have audio focus.
  Promise<void> abandon();
};
```

**Comment by tguilbert-google** [comment]
> A site may have both alert sounds ("transient") and a music stream ("playback").
+1 one to this question/issue/answer. I support adding a way to allow transient sounds from sites with longer playback.
